mongo/mongo.go

```diff
@@ -65,27 +86,26 @@ func (m *Mongo) Checker(ctx context.Context, state *healthcheck.CheckState) erro
 // GetCollections retrieves all collection documents
 func (m *Mongo) GetCollections(ctx context.Context, queryParams collections.QueryParams) ([]models.Collection, int, error) {
 
-	s := m.Session.Copy()
-	defer s.Close()
-
-	var q *mgo.Query
-	var query interface{}
+	var q *dpMongoDriver.Find
+	query := bson.D{}
 
 	if len(queryParams.NameSearch) > 0 {
-		query = bson.M{"name": &bson.RegEx{Pattern: queryParams.NameSearch, Options: "i"}}
+		query = bson.D{{"name", primitive.Regex{Pattern: queryParams.NameSearch, Options: "i"}}}
 	}
 
-	q = s.DB(m.Database).C(m.CollectionsCollection).Find(query)
+	q = m.Connection.
+		GetConfiguredCollection().
+		Find(query)
 
 	switch queryParams.OrderBy {
 	case collections.OrderByPublishDate:
-		q.Sort("publish_date")
+		q.Sort(bson.D{{"publish_date", 1}})
 	}
 
-	totalCount, err := q.Count()
+	totalCount, err := q.Count(ctx)
 	if err != nil {
 		log.Error(ctx, "error getting count of collections from mongo db", err)
-		if err == mgo.ErrNotFound {
+		if dpMongoDriver.IsErrNoDocumentFound(err) {
```

Comment:
count will not return an error here

Reply:
Not sure exactly what you mean by this - are you suggesting that count will not return a `ErrNoDocumentFound` error?

Reply:
I have assumed this is the case and removed the specific error checking